### DISCUSSION TITLE
Accessing req.body from getServerSideProps

### DISCUSSION THREAD

**Discussion by dkprog** [vercel/next.js#14979] [Help] [answered]
I'm trying to render a page based on previously submitted form data via POST.

I can use req.body when working with API however, but I unable to render a page from there.

How can I use req.body inside getServerSideProps?

**Comment by adrianfalleiro** [comment, marked as answer]
Hacky, but here is how I'm doing it in a project of mine.

```js
import { parseBody } from 'next/dist/next-server/server/api-utils';

export const getServerSideProps = async (context) => {
  const body = await parseBody(context.req, '1mb');

  //...
}
```

**Reply by dkprog** [reply]
Thank you!

Do you know why is this unavailable inside getServerSideProps? Does the authors have anything against using it this way to create a MPA?

**Reply by GioLogist** [reply]
@adrianfalleiro nice workaround!

Just a side note, this only worked for me after also updating `next.config.js` with the following:

```js
module.exports = {
  webpack: (config, { isServer }) => {
    // Fixes npm packages that depend on `fs` module
    if (!isServer) {
      config.node = {
        fs: 'empty',
      };
    }

    return config;
  },
};
```

**Reply by sidmohanty11** [reply]
In next v12, the import is 
```javascript
import { parseBody } from 'next/dist/server/api-utils'
```

**Reply by zakiego** [reply]
In next 12.2
```javascript
import { parseBody } from "next/dist/server/api-utils/node";

export async function getServerSideProps(context) {
  const body = await parseBody(context.req, "1mb");
  console.log(body);
  return { props: {} };
}
```

**Reply by Seb-C89** [reply]
In next ~v13, the import is
```js
import { parseBody } from "next/dist/server/api-utils/node/parse-body.js"
```

**Comment by lvpaspaltzis** [comment]
For future references:

```typescript
import getRawBody from 'raw-body';

...

export const getServerSideProps: GetServerSideProps = async ({req}) => {
  if (req.method == "POST") {
    const body = await getRawBody(req)
    console.log(body.toString("utf-8"))
  }
  
  return { props: {} };
};
```

**Reply by isekaimaou1109** [reply]
@dracoboros it worked as a charm. thanks

**Comment by xav-ie** [comment]
I am very confused... I am trying to fix the types of my repo, and found I was making some sneaky assumptions in a .d.ts file. I have since converted to .ts, but it seems like .body is not allowed... except it is. I was able to use .body with .d.ts, by forcing the type to allow it, but I am now confused as to why this is not on the type and there has to be body parsing...

> I'm trying to render a page based on previously submitted form data via POST.
> 
> I can use req.body when working with API however, but I unable to render a page from there.
> 
> How can I use req.body inside getServerSideProps?
